src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -48,29 +36,26 @@ protected function configureRoutes(RouteCollectionBuilder $routes)
      *         ->controller('App\Controller\AdminController::dashboard')
      *     ;
      */
-    protected function configureRouting(RoutingConfigurator $routes): void
-    {
-        @trigger_error(sprintf('Not overriding the "%s()" method is deprecated since Symfony 5.1 and will trigger a fatal error in 6.0.', __METHOD__), E_USER_DEPRECATED);
-    }
+    abstract protected function configureRoutes(RoutingConfigurator $routes);
 
     /**
      * Configures the container.
      *
      * You can register extensions:
      *
-     *     $c->loadFromExtension('framework', [
+     *     $c->extension('framework', [
      *         'secret' => '%secret%'
      *     ]);
      *
      * Or services:
      *
-     *     $c->register('halloween', 'FooBundle\HalloweenProvider');
+     *     $c->services()->set('halloween', 'FooBundle\HalloweenProvider');
      *
      * Or parameters:
      *
-     *     $c->setParameter('halloween', 'lot of fun');
+     *     $c->parameters()->set('halloween', 'lot of fun');
      */
-    abstract protected function configureContainer(ContainerBuilder $c, LoaderInterface $loader);
+    abstract protected function configureContainer(ContainerConfigurator $c);
```

Comment:
That's also a BC break. We would need a new trait.

Reply:
it's not a BC break because this is a trait: everyone is forced to override it right now.

Reply:
But isn't it an issue if you override it with a different signature ?

Reply:
I was surprised at first too, but the trait does not enforce a signature like an abstract class. And this makes sense when you think about it, its definition is just replaced and always has. Good for reusable code not defining contract.